src/js/components/Select/SelectContainer.js

```diff
@@ -36,22 +40,58 @@ const SelectOption = styled(Button)`
   width: 100%;
 `;
 
-const ClearButton = ({ clear, onClear, name, theme, setFocus }) => {
+const ClearButtonContainer = styled(Box)``;
+const ClearButtonText = styled(Text)``;
+const ClearButtonComponent = ({
+  className,
+  clear,
+  onClear,
+  name,
+  theme,
+  setFocus,
+}) => {
   const { label, position } = clear;
   const align = position !== 'bottom' ? 'start' : 'center';
   const buttonLabel = label || `Clear ${name || 'selection'}`;
+
   return (
     <Button
+      // className must be passed for styled-components component interpolation
+      // https://styled-components.com/docs/advanced#referring-to-other-components
+      className={className}
       onClick={onClear}
       onFocus={() => setFocus(true)}
       onBlur={() => setFocus(false)}
     >
-      <Box {...theme.select.clear.container} align={align}>
-        <Text {...theme.select.clear.text}>{buttonLabel}</Text>
-      </Box>
+      <ClearButtonContainer {...theme.select.clear.container} align={align}>
+        <ClearButtonText {...theme.select.clear.text}>
+          {buttonLabel}
+        </ClearButtonText>
+      </ClearButtonContainer>
     </Button>
   );
 };
+const ClearButton = styled(ClearButtonComponent)`
+  &:hover {
+    ${(props) =>
+      props.theme.select.clear.hover?.background &&
+      `${ClearButtonContainer} {
```

Comment:
Since the ClearButtonContainer fills its parent ClearButton, can we just put this hover styling directly on ClearButtonContainer instead?